corehq/apps/app_manager/views/apps.py

```diff
@@ -492,12 +489,17 @@ def load_app_from_slug(domain, username, slug):
                                                username=username)
                         multimedia.add_domain(domain, owner=True)
                         app.create_mapping(multimedia, MULTIMEDIA_PREFIX + path)
+    return _build_sample_app(app)
+
 
-    comment = _("A sample application you can try out in Web Apps")
-    build = make_build(app, allow_prune=False, comment=comment)
-    build.is_released = True
-    build.save(increment_version=False)
-    return build
+def _build_sample_app(app):
+    errors = app.validate_app()
+    if not errors:
+        comment = _("A sample application you can try out in Web Apps")
+        copy = app.make_build(comment=comment)
+        copy.is_released = True
+        copy.save(increment_version=False)
+        return copy
```

Comment:
This change inadvertently fixes https://dimagi-dev.atlassian.net/browse/HI-788 - in the old version of this code, `make_build` can return `None` (if the app fails to validate).  That appears to be what happens.  Still not sure why that fails though.

Reply:
Why are template apps failing to validate? Or is that what you're not sure of?